### PR TITLE
Convert more audio props, check for non-convertable extensions, add more names found by @A-xesey

### PR DESCRIPTION
**Pull request by 0KepOnline**

### CODE REVIEW

File: src/sporemodder/view/syntax/XmlSyntax.java [hunk 1]
```diff
@@ -101,7 +101,10 @@ public class XmlSyntax implements SyntaxFormatFactory {
 	public boolean isSupportedFile(File file) {
 		try {
 			// Unfortunately, this syntax highlighter is too slow for "big" files
-			return file.getName().endsWith(".xml") && Files.size(file.toPath()) < 10240;
+			return Files.size(file.toPath()) < 10240 && (
+				file.getName().endsWith(".xml") |
+				file.getName().endsWith(".eapdPixie")
+			);
```
Comment (emd4600) [inline src/sporemodder/view/syntax/XmlSyntax.java:107]: I think it's better to revert the conditions. Checking file size might be slower, no need to do it if it is not one of the extensions.

**Review by emd4600 — CHANGES_REQUESTED** [review]

File: src/sporemodder/view/syntax/XmlSyntax.java [hunk 2]
```diff
@@ -101,7 +101,11 @@ public class XmlSyntax implements SyntaxFormatFactory {
 	public boolean isSupportedFile(File file) {
 		try {
 			// Unfortunately, this syntax highlighter is too slow for "big" files
-			return file.getName().endsWith(".xml") && Files.size(file.toPath()) < 10240;
+			return (
+				file.getName().endsWith(".xml") |
+				file.getName().endsWith(".eapdPixie") &
+				Files.size(file.toPath()) < 10240
+			);
```
Comment (emd4600) [inline src/sporemodder/view/syntax/XmlSyntax.java:108]: The condition should use boolean operators (`||`, `&&`), not binary ones (`|`, `&`). Also, the name conditions should be in parenthesis. Like this:
```java
(file.getName().endsWith(".xml") || file.getName().endsWith(".eapdPixie")) && Files.size(file.toPath()) < 10240
```

**Review by emd4600 — CHANGES_REQUESTED** [review]

File: src/sporemodder/view/syntax/XmlSyntax.java (hunk 2 above)
Comment (0KepOnline) [reply src/sporemodder/view/syntax/XmlSyntax.java:107]: Done.

File: src/sporemodder/file/ResourceKey.java [hunk 3]
```diff
@@ -103,25 +103,48 @@ public void writeBE(StreamWriter stream) throws IOException {
...
+	 * @param hexOnly If true, the returned string will only have hexadecimal IDs with no conversion.
+	 */
+	public String toFormattedString(char groupSeparator, char typeSeparator, boolean quotesIfNeeded, boolean hexOnly) {
 		HashManager hasher = HashManager.get();
 		StringBuilder sb = new StringBuilder();
 		
-		if (groupID != 0) {
-			sb.append(hasher.getFileName(groupID));
-			sb.append('!');
+		if (hexOnly == false) {
```
Comment (emd4600) [inline src/sporemodder/file/ResourceKey.java:116]: Use `!hexOnly` instead of `hexOnly == false`

File: src/sporemodder/file/ResourceKey.java [hunk 4]
```diff
@@ -103,25 +103,48 @@ public void writeBE(StreamWriter stream) throws IOException {
...
+			}
+			sb.append(hasher.hexToStringUC(instanceID));
+			if (typeID != 0) {
+				sb.append(typeSeparator);
+				sb.append(hasher.hexToStringUC(typeID));
+			}
 		}
 		
+		
+		if (sb.toString().indexOf(' ') != -1 && quotesIfNeeded == true) return '"' + sb.toString() + '"';
```
Comment (emd4600) [inline src/sporemodder/file/ResourceKey.java:140]: Use `quotesIfNeeded` instead of `quotesIfNeeded == true`

**Review by emd4600 — CHANGES_REQUESTED** [review]

File: src/sporemodder/file/ResourceKey.java (hunk 4 above)
Comment (0KepOnline) [reply src/sporemodder/file/ResourceKey.java]: Done.

**Review by emd4600 — APPROVED** [review]

**Pull request merged**